### DISCUSSION TITLE
Missing Entities

### DISCUSSION THREAD

**Discussion by querys1** [SukramJ/aiohomematic#729] [Q&A]
I made an Update for HA and  afterwards I hade some issues with setting up the Homematic Addon again. But finally it worked.

Unfortunately some of the entities are missing, which I had before.

They are all related to the HMIP FAL230 Floorheating Actor. It was possible to get the states if a heating circle is on or off. With this information I drawed nice diagramms to get an overview, if the heating is correctly and how long they are working.

Unfortunatley these entities are missing now. Any suggestion, how to get them back?
This is how the diagramm looks like. The one which seems to be available grabbed a wrong entity.

![example](https://user-images.githubusercontent.com/116440848/205491358-c3152310-532b-41a1-8208-36a88e5bfafe.jpg)

The FAL230 does only offer some 3 entities now (Voltage, RSSI, operation mode).

Thanks a lot for any ideas.

**Comment by SukramJ** [comment]
Since you are withholding the not inconsiderable information from us, what kind of entities are involved, i can unfortunately only guess. 
I would see if you just have to activate the corresponding entities. See documentation.

**Comment by querys1** [comment]
Sorry I do not want to hold back any information.
Before the following entities were available

binary_sensor.hmip_fal230_c6_erdgeschoss_state_ch1

And all other channels (ch2, ch3, ch4…..)

How do I activate the entities?
Configurations—> devices —> entities —> show all?
If yes this is the way how I activated everything which is available.
I was wondering because all the binaries are gone only sensors are available.
[config_entry-homematicip_local-5db11752605636bc471f3f3b64f4d3b5.json.txt](https://github.com/danielperna84/hahomematic/files/10148642/config_entry-homematicip_local-5db11752605636bc471f3f3b64f4d3b5.json.txt)

**Reply by SukramJ** [reply]
> How do I activate the entities?

As described in the docs:
# Deactivated Entities
A lot of additional entities were initially created deactivated and can be activated, if necessary, in the advanced settings of the entity.

**Reply by querys1** [reply]
okay thanks I already figured that out, but I do not have any deactivated entities. I do not have any binary sensor for Homematic at all.
Is there any possibility to add the binary sensors manually or force the addon to search for them? I tried to reboot HA and restarted the addon, but the entities is still as before (286 items)

can I remove and re-add the FAL230?

**Reply by SukramJ** [reply]
Best option is to remove the integration and add it again.

If the problem still exists your best option would be to create an issue here and add ALL the requested information.

**Reply by SukramJ** [reply]
Ensure that, after the removal of the integration, the directory `"your HA config"/homematicip_local/cache` is empty.

**Reply by querys1** [reply]
Thanks for the help.
I tried reinstalling, but with no help. Afterwards I downgraded to the previous version (1.21.3) , and now the missing entities are back again.
[config_entry-homematicip_local-924e2004b1f743a5d096170c62628b99.json.txt](https://github.com/danielperna84/hahomematic/files/10148948/config_entry-homematicip_local-924e2004b1f743a5d096170c62628b99.json.txt)

**Reply by querys1** [reply]
In total I have now 330 entities (instead of 286 on 1.22.0
